samtranslator/swagger/swagger.py

```diff
@@ -285,6 +286,107 @@ def _make_cors_allowed_methods_for_path(self, path):
+
+    def set_path_default_authorizer(self, path, default_authorizer, authorizers):
+        """
+        Sets the DefaultAuthorizer for each method on this path. The DefaultAuthorizer won't be set if an Authorizer
+        was defined at the Function/Path/Method level
+
+        :param string path: Path name
+        :param string default_authorizer: Name of the authorizer to use as the default. Must be a key in the authorizers param.
+        :param list authorizers: List of Authorizer configurations defined on the related Api.
+        """
+        for method_name, method in self.paths[path].items():
+            self.set_method_authorizer(path, method_name, default_authorizer, authorizers, default_authorizer=default_authorizer, is_default=True)
+
+    def add_auth_to_method(self, path, method_name, auth, api):
+        """
+        Adds auth settings for this path/method. Auth settings currently consist solely of Authorization
+        (aka Custom Authorizers) but this method will eventually include setting other auth settings such as API Key,
+        Resource Policy, etc.
+
+        :param string path: Path name
+        :param string method_name: Method name
+        :param dict auth: Auth configuration such as Authorizers, ApiKey, ResourcePolicy (only Authorizers supported
+                          currently)
+        :param dict api: Reference to the related Api's properties as defined in the template.
+        """
+        method_authorizer = auth and auth.get('Authorizer')
+        if method_authorizer:
+            api_auth = api.get('Auth')
+            api_authorizers = api_auth and api_auth.get('Authorizers')
+            default_authorizer = api_auth and api_auth.get('DefaultAuthorizer')
+
+            self.set_method_authorizer(path, method_name, method_authorizer, api_authorizers, default_authorizer)
+
+    def set_method_authorizer(self, path, method_name, authorizer_name, authorizers, default_authorizer,
+                              is_default=False):
+        existing_security = self.paths[path][method_name].get('security', [])  # TEST: [{'sigv4': []}, {'api_key': []}])
+        authorizer_names = set(authorizers.keys())
+        existing_non_authorizer_security = []
+        existing_authorizer_security = []
+
+        # Split existing security into Authorizers and everything else
+        # (e.g. sigv4 (AWS_IAM), api_key (API Key/Usage Plans), NONE (marker for ignoring default))
+        # We want to ensure only a single Authorizer security entry exists while keeping everything else
+        for security in existing_security:
+            if authorizer_names.isdisjoint(security.keys()):
+                existing_non_authorizer_security.append(security)
+            else:
+                existing_authorizer_security.append(security)
+
+        none_idx = -1
+        authorizer_security = []
+
+        # If this is the Api-level DefaultAuthorizer we need to check for an
```

Comment:
This implementation isn't ideal. The existing order of operations is:

1. Apply all Function Event configuration
1. Apply API configuration

This makes it hard to apply a DefaultAuthorizer, especially when 'NONE' is defined on the Function Event. Current logic is to add a 'NONE' entry to the Swagger and then remove it when DefaultAuthorizer comes through. Similarly, DefaultAuthorizer won't be added if it detects an existing entry in `security` which matches a key defined in `Api.Auth.Authorizers`.

Ideally we would add DefaultAuthorizer first and then remove/override as needed, but I'd rather not complicate the order of operations (unless there's already precedent of that which I'm not aware of).